src/index.ts

```diff
@@ -51,7 +53,7 @@ function dryRun(uploadHost, token, query, uploadFile, source) {
  * @param {string} args.feature Toggle features
  * @param {string} args.source Track wrappers of the uploader
  */
-async function main(args) {
+export async function main(args: UploaderArgs): Promise<void | Record<string, unknown>> {
```

Comment:
Is the return type correct? I was under the impression that it was supposed to return a `Promise<void>`. The CLI script does not handle any return values.

Reply:
Yes, it's used here https://github.com/codecov/uploader/blob/typescript-migration/test/index.test.ts#L48-L70

Reply:
That test looks like it's testing functionality that isn't used anywhere else.

Reply:
It is an "implementation" test. Testing end to end of `main` to insure nothing catches fire. https://github.com/codecov/uploader/issues/16 is  be desired behavior, the above test was written as a last check before we wasted time building binaries.